### ISSUE TITLE
JS variant throws TypeError exception with AMF 5.0.1

### ISSUE THREAD

**Issue by deiteris** [aml-org/amf#1222]
Your issue may already be reported! Please search on Github issues before creating one.

* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] question
  - [ ] other

* **What is the current behavior?**

I've updated from AMF 5.0.0 to 5.0.1 with `npm`, and now AMF parser throws an exception with the following stack trace:
```sh
function Uta(a,b){var c=new Yxa;c.eMa=a;c.fMa=b;c.ek=(new sp).a();c.hJa=(new Tw).Rl(t(function(){return function(e){e=(new Eb).f(e.ta.Vw);return Tc(e.Vd)}}(c)));c.asa=(new Uw).a();c.QJa=(new Zk).a();return c}function iya(a,b,c){b=b.dc;var e=K();b=b.Eh(e.u);a=t(function(f,g){return function(h){if(null!==h){var k=h.Ca();h=Vw(h);var l=bya(k);h=aya(f,g,l.b()?""+h:l.c());return dya(f,k,h,(fya(f),!1))}throw(new v).e(h);}}(a,c));c=K();return b.ra(a,c.u)}function hya(a){return(Cg(a)?jya():kya()).jJa(a)}
                                                              ^

TypeError: (intermediate value).a is not a function
    at new Uta (E:\Sources\amf-gh-976\node_modules\amf-client-js\amf.js:20606:63)
    at Object.<anonymous> (E:\Sources\amf-gh-976\node_modules\amf-client-js\amf.js:1392:172)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:94:18)
    at Object.<anonymous> (E:\Sources\amf-gh-976\amf-parser.js:1:13)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
```

* **If the current behavior is a bug, please provide the minimal steps to reproduce with AMF code. Please don't upload AMF code mixed with yours.**

Install AMF 5.0.1 from `npm` and run the following code:
```js
const amf = require('amf-client-js');
amf.RAMLConfiguration.RAML10().baseUnitClient();
```

* **What is the expected behavior?**

No exception error.

* **Please tell us about your environment:**
  
  - AMF Version: 5.0.1
  - AMF Distribution: JS 
  - Operating System: Windows 10 21H1

**Comment by looseale** [comment]
Hi @deiteris, we had a publication issue with the version 5.0.1. We already released a HF, 5.0.1-0 that should fix that error.
Please try it using that version and let us know if it works.

**Comment by deiteris** [comment]
Hi @looseale. I see, this release wasn't listed on the github and `^` in the version doesn't upgrade to it. It works with `5.0.1-0`, thanks!
